src/webgl/material.js

```diff
@@ -1175,6 +1175,84 @@ p5.prototype.shininess = function (shine) {
   return this;
 };
 
+/**
+ * Sets the metalness property of a material used in 3D rendering.
+ *
+ * The metalness property controls the degree to which the material
+ * appears metallic. A higher metalness value makes the material look
+ * more metallic, while a lower value makes it appear less metallic.
+ *
+ * The default and minimum value is 0, indicating a non-metallic appearance.
+ *
+ * Unlike other materials, metals exclusively rely on reflections, particularly
+ * those produced by specular lights (mirrorLike lights). They don't incorporate
+ * diffuse or ambient lighting. Metals use a fill color to influence the overall
+ * color of their reflections. Pick a fill color, and you can easily change the
+ * appearance of the metal surfaces. When no fill color is provided, it defaults
+ * to using white color.
+ *
+ * @method metalness
+ * @param {Number} metallic - The degree of metalness.
+ * @example
+ * <div class="notest">
+ * <code>
+ * let img;
+ * let slider;
+ * function preload() {
+ *   img = loadImage('assets/outdoor_spheremap.jpg');
+ * }
+ * function setup() {
+ *   createCanvas(100, 100, WEBGL);
+ *   slider = createSlider(0, 400, 100, 1);
+ *   slider.position(0, height);
+ * }
+ * function draw() {
+ *   background(220);
+ *   imageMode(CENTER);
+ *   push();
+ *   translate(0, 0, -200);
+ *   scale(2);
+ *   image(img, 0, 0, width, height);
+ *   pop();
+ *   imageLight(img);
+ *   fill('green');
+ *   specularMaterial('gray');
+ *   shininess(slider.value());
+ *   metalness(100);
+ *   noStroke();
+ *   scale(2);
+ *   sphere(15);
+ * }
+ * </code>
+ * </div>
+ * @example
+ * <div>
+ * <code>
+ * function setup() {
+ *   createCanvas(100, 100, WEBGL);
+ * }
+ * function draw() {
+ *   noStroke();
+ *   background('black');
+ *   fill(255, 215, 0);
+ *   pointLight(255, 255, 255, 200, 150, 8000);
+ *   pointLight(255, 255, 255, 5000, 5000, 75);
+ *   specularMaterial('gray');
+ *   ambientLight(255);
+ *   shininess(2);
+ *   metalness(100);
```

Comment:
It would be great for one of the examples to have a slider that controls metalness so people can see the effect it has. Maybe this one would be good for that? A quick update to this that I was trying out:
```js
let slider
function setup() {
  createCanvas(100, 100, WEBGL);
  slider = createSlider(0, 200, 100);
}
function draw() {
  noStroke();
  background(100);
  fill(255, 215, 0);
  pointLight(255, 255, 255, 5000, 5000, 75);
  specularMaterial('gray');
  ambientLight(100);
  shininess(2);
  metalness(slider.value());
  sphere(45);
}
```

The things I was hoping to improve upon were:
- added a slider to control metalness
- a different background color so you can see the outline of the shape better
- lights that show the form of the shape a bit more clearly (it's definitely still not perfect, any ideas on how to improve this? maybe making the light orbit the sphere over time? maybe using a torus instead of a sphere?)
- less ambient light so the non-metal version is less blown out and you can see the highlights better